src/main.cpp

```diff
@@ -11,11 +12,19 @@ int main(int argc, char *argv[]){
     }
 
     // set the length of the character set
-    if(argv[1] == "-L")
+    int length = 0;
+    for(int i = 0; i<argc; i++)
     {
-        int length = atoi(argv[2]);
-        // generate length using length generator.
-
+        if(strcmp(argv[i], "-L") == 0)
+        {
+            // And since the length entered by the user may be after '-S' so we
+            // have to take another loop to identify where it is.
+            for(int j = i; j<argc; j++)
```

Comment:
There is no need for this loop. If -L is in position i then value will be in position i+1